tests/SwishPaymentTest.php

```diff
@@ -125,10 +141,279 @@ public function testCreateSaleTransaction($paymentId)
 
         $result = $responseService->getResponseData();
 
-        $this->assertTrue(is_array($result));
-        $this->assertTrue(isset($result['payment']));
-        $this->assertTrue(isset($result['sale']));
-        $this->assertTrue(isset($result['sale']['transaction']));
-        $this->assertTrue(($result['sale']['transaction']['type']) === 'Sale');
+        $this->assertIsArray($result);
+        $this->assertArrayHasKey('payment', $result);
+        $this->assertArrayHasKey('sale', $result);
+        $this->assertArrayHasKey('id', $result['sale']);
+        $this->assertArrayHasKey('transaction', $result['sale']);
+        $this->assertArrayHasKey('type', $result['sale']['transaction']);
+        $this->assertArrayHasKey('state', $result['sale']['transaction']);
+        $this->assertEquals('Sale', $result['sale']['transaction']['type']);
+        $this->assertEquals('AwaitingActivity', $result['sale']['transaction']['state']);
+
+        // Wait for state change AwaitingActivity -> Ready
+        $saleId = $result['sale']['id'];
+        $attempts = 0;
+
+        do {
+            sleep(5);
```

Comment:
Codacy found an issue: [The use of function sleep() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=6123563)